verticapy/tests_new/machine_learning/vertica/__init__.py

```diff
@@ -17,48 +17,497 @@
-    "RandomForestClassifier": 1e-0,
-    "DecisionTreeRegressor": 1e-0,
-    "DecisionTreeClassifier": 1e-0,
-    "DummyTreeRegressor": 1e-01,
-    "DummyTreeClassifier": 1e-01,
-    "XGBRegressor": 1e-0,
-    "XGBClassifier": 1e-0,
-    "Ridge": 1e-02,
-    "Lasso": 1e-02,
-    "ElasticNet": 1e-02,
-    "LinearRegression": 1e-02,
-    "LinearSVR": 1e-01,
-    "PoissonRegressor": 1e-01,
-    "AR": 1e-01,
-    "MA": 1e-01,
-    "ARMA": 1e-02,
-    "ARIMA": 1e-00,
-    "TENSORFLOW": 1e-06,
+rel_abs_tol_map = {
+    "RandomForestRegressor": {
+        "explained_variance": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "max_error": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "median_absolute_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_absolute_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_squared_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_squared_log_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "rmse": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        **dict.fromkeys(["r2", "R-squared"], {"rel": 1e-01, "abs": ABS_TOLERANCE}),
+        **dict.fromkeys(
+            ["r2_adj", "Adj. R-squared"], {"rel": 1e-01, "abs": ABS_TOLERANCE}
+        ),
+        "aic": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "bic": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "F-statistic": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "Prob (F-statistic)": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "Kurtosis": {"rel": 1e-02, "abs": ABS_TOLERANCE},
+        "Skewness": {"rel": 1e-03, "abs": ABS_TOLERANCE},
+        "Jarque-Bera (JB)": {"rel": 1e-02, "abs": ABS_TOLERANCE},
+        "df": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "ss": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "ms": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "f": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "p_value": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "coef_": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "intercept_": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "score": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "predict": {"rel": 1e-02, "abs": ABS_TOLERANCE},
+        "to_python": {"rel": REL_TOLERANCE, "abs": ABS_TOLERANCE},
+        "load_model": {"rel": 1e-02, "abs": ABS_TOLERANCE},
+    },
+    "RandomForestClassifier": {
+        "auc": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "prc_auc": {"rel": 1e-02, "abs": ABS_TOLERANCE},
+        "accuracy": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "log_loss": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "precision": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "recall": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "f1_score": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "mcc": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "markedness": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "csi": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "confusion_matrix": {"rel": 1e-00, "abs": ABS_TOLERANCE},
```

Comment:
Are you comparing all the 4 parts of the matrix: TP, FP, TN, FN?

Reply:
yes.